src/main.cpp

```diff
@@ -7163,13 +7163,20 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
 
     else if (strCommand == "inv")
     {
-        vector<CInv> vInv;
-        vRecv >> vInv;
-        if (vInv.size() > MAX_INV_SZ)
+        auto vRecvStreamSz = ReadCompactSize(vRecv);
+        if (vRecvStreamSz > MAX_INV_SZ)
         {
             Misbehaving(pfrom->GetId(), 20);
-            return error("message inv size() = %u", vInv.size());
+            return error("message inv size() = %u", vRecvStreamSz);
+        }
+        else if(vRecvStreamSz * sizeof(CInv)!= vRecv.in_avail()) {
+            Misbehaving(pfrom->GetId(), 20);
+            return error("invalid message payload for %u", vRecvStreamSz);
         }
```

Comment:
If I understand correctly, the meaning of this check is to increase the ban score in case the declared size is different from the real size.

At first glance, it makes sense to me, even though it is a different behavior compared to the past.

That said, do you think it's possible for a message sent by an honest peer to reach this branch of code and get banned (for instance because of corrupted data at the network level)?

Reply:
IMO very little. There would have to be a network corruption on tcp level.